### PR TITLE
refactor: code around waiting in tests

### PR DESCRIPTION
**Pull request by caldrian**
This PR refactors code around waiting in tests.

1. unclear code like this is replaced
```
wait(timeout: 0.5) {
    await quickSyncObserver.waitForQuickSyncToFinish()
}
```

2. simple waiting for some time is done like this now
```
wait(for: [XCTestExpectation().inverted()], timeout: 0.05)
```
instead of `OperationStatusTests.performRunLoopTick()` or similar hacks

3. waiting for a condition to evaluate to true can be done like this
```
wait(forConditionToBeTrue: message.isExpired, timeout: 2)
```

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Message Sending/MessageDependencyResolverTests.swift [hunk 1]
```diff
@@ -32,9 +32,12 @@ final class MessageDependencyResolverTests: MessagingTestBase {
             .arrange()
 
         // then test completes
-        wait(timeout: 0.5) {
+        let expectation = XCTestExpectation(description: "action is done within 500ms")
+        Task {
             try await messageDependencyResolver.waitForDependenciesToResolve(for: message)
+            expectation.fulfill()
         }
+        wait(for: [expectation], timeout: 0.5)
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Message Sending/MessageDependencyResolverTests.swift:40]: Thank you! 💪

File: wire-ios-sync-engine/Tests/Source/Integration/SlowSyncTests+NotificationsV3.swift [hunk 2]
```diff
@@ -60,9 +61,7 @@ class SlowSyncTests_NotificationsV3: IntegrationTest {
         self.performQuickSync()
 
         // THEN
-        let result = wait(withTimeout: 1) {
-            self.userSession?.applicationStatusDirectory.syncStatus.isSlowSyncing == true
-        }
-        XCTAssertTrue(result, "it should perform slow sync")
+        // ensure it performs slow sync
+        wait(forConditionToBeTrue: self.userSession?.applicationStatusDirectory.syncStatus.isSlowSyncing == true, timeout: 5)
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Tests/Source/Integration/SlowSyncTests+NotificationsV3.swift:65]: Quite deep nested test.. maybe the unit test should rather test smaller parts?
But okay for the topic here I understand not to touch it 😁

File: wire-ios-sync-engine/Tests/Source/Calling/MLSConferenceStaleParticipantsRemoverTests.swift [hunk 3]
```diff
@@ -150,8 +152,7 @@ class MLSConferenceStaleParticipantsRemoverTests: MessagingTest {
         }
 
         // set expectations
-        let expectation = XCTestExpectation()
-        expectation.isInverted = true
+        let expectation = InvertedTestExpectation()
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Tests/Source/Calling/MLSConferenceStaleParticipantsRemoverTests.swift]: Hmm I prefer the Apple version, but it's personal taste..

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-sync-engine/Tests/Source/Calling/MLSConferenceStaleParticipantsRemoverTests.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios-sync-engine/Tests/Source/Calling/MLSConferenceStaleParticipantsRemoverTests.swift]: Changed it, what do you think now?
53358e05f9adb868b1042078d233044b2a3c0376

File: wire-ios-sync-engine/Tests/Source/Integration/SlowSyncTests+NotificationsV3.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios-sync-engine/Tests/Source/Integration/SlowSyncTests+NotificationsV3.swift:65]: I agree

**Review by johnxnguyen — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Pull request merged**